src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -1,17 +1,144 @@
 import { PropTypes } from 'prop-types';
 import {
-  ActionRow,
+  ActionRow, Alert,
   Button,
+  Form,
   Icon,
   IconButtonWithTooltip,
   ModalDialog,
   useToggle,
 } from '@openedx/paragon';
-import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
+import { DeleteOutline, EditOutline, Info } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance } from '../hooks';
+import { useClearRequest, useRequestError } from '../../../data/redux/hooks';
+import * as constants from '../../../data/constants';
+
+const EditModal = (isOpen, close, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const [additionalTimeError, setAdditionalTimeError] = useState(false);
+  const requestError = useRequestError(constants.RequestKeys.createAllowance);
+  const resetRequestError = useClearRequest(constants.RequestKeys.createAllowance);
+
+  const initialFormState = {
+    username: allowance.username,
+    'exam-id': allowance.examId,
+    'exam-name': allowance.examName,
+  };
+  const [form, setForm] = useState(initialFormState);
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    resetRequestError();
+    setForm(prev => ({ ...prev, [name]: value }));
+  };
+
+  const onClose = () => {
+    resetRequestError();
+    setForm(initialFormState);
+    close();
+  };
+
+  const onSubmit = () => {
+    const extraTimeMins = +form['extra-time-mins'] || 0;
+    // todo: We should maybe move the handling of this validation to the backend,
```

Comment:
Added this todo, as this is a separate ticket I created here: https://2u-internal.atlassian.net/browse/COSMO-446

Reply:
Agree with the ticket. Validation should be done on both the frontend and the backend.